gravitee-am-repository/gravitee-am-repository-jdbc-api/src/main/java/io/gravitee/am/repository/jdbc/provider/impl/R2DBCConnectionProvider.java

```diff
@@ -152,22 +163,29 @@ public ConnectionProvider stop() throws Exception {
         if (this.oauthConnectionFactory != null) {
             ((R2DBCPoolWrapper) this.oauthConnectionFactory).shutdown();
         }
+        if (this.gatewayConnectionFactory != null) {
+            ((R2DBCPoolWrapper) this.gatewayConnectionFactory).shutdown();
+        }
         return this;
     }
 
     @Override
     public void afterPropertiesSet() throws Exception {
+        var useOAuth2SettingsOld = environment.getProperty("oauth2.use-management-settings", boolean.class, true);
+        useGatewaySettings = environment.getProperty("gateway.use-management-settings", boolean.class, useOAuth2SettingsOld);
```

Comment:
**issue**: this need to be consistent with the mongo implementation.

Let's go with the mongo implementation for now

```
        final var useOAuth2Settings = environment.getProperty("oauth2.use-management-settings", Boolean.class, true);
        final var useGatewaySettings = environment.getProperty("gateway.use-management-settings", Boolean.class, true);
        notUseOAuth2Settings = !useOAuth2Settings;
        notUseGatewaySettings = !useGatewaySettings;
```

Reply:
maybe this one if you accept the proposal made on Mongo side
```
        final var useMngSettingsForOauth2 = environment.getProperty("oauth2.use-management-settings", Boolean.class, true);
        final var useMngSettingsForGateway = environment.getProperty("gateway.use-management-settings", Boolean.class, true);
        notuUseMngSettingsForOauth2 = !useMngSettingsForOauth2;
        notUseMngSettingsForGateway = !useMngSettingsForGateway;
```

Reply:
fixed